### ISSUE TITLE
Apollo watchQuery returns no results or errors

### ISSUE THREAD

**Issue by pwharff** [apollographql/apollo-angular#1334]
**CONTEXT**

I decided to wrap up the `GraphQLModule` that is created by `ng add apollo-angular` into our own Angular Shared Module. 

Here is the module that is generated by `ng add apollo-angular` cli:

```
const uri = 'graphql';
export function createApollo(httpLink: HttpLink) {
  return {
    link: httpLink.create({uri}),
    cache: new InMemoryCache(),
  };
}

@NgModule({
  exports: [ApolloModule, HttpLinkModule],
  providers: [
    {
      provide: APOLLO_OPTIONS,
      useFactory: createApollo,
      deps: [HttpLink],
    },
  ],
})
export class GraphQLModule {}
```

We've made minor changes to this module and wrapped it up in our own NPM package. And added it to the imports array of `app.module.ts`

**TL;DR**

While performing queries inside of this module (for auth purposes), I properly receive responses from the server. 

However, when using this module in another application, we receive no response or errors: 

```
  myQuery = gql`
    {
      member {
        id
        details {
          id
        }
      }
    }
  `

  queryOptions: WatchQueryOptions = {query: this.myQuery, errorPolicy: 'all'}
  myQueryObs: Observable<any>

  constructor(private apollo: Apollo) {}

  ngOnInit() {
    this.myQueryObs = this.apollo.watchQuery<Query>(this.queryOptions)
    .valueChanges.pipe(map (results => {
      console.log('RESULTS', results) // NEVER GET HERE
    }));
  }
```

I've logged the crap out of it and cannot determine why it is not responding at all. It acts like Apollo is not instantiated at all, but console output proves otherwise. 

Any advice or help is greatly appreciated!

**Comment by kamilkisiela** [comment]
I can't help you without any reproduction. I don't know if it's a lazy loaded module and any other details.

**Comment by pwharff** [comment]
Ok thanks, let push to a repo and create a StackBlitz... Give me a moment or two or ten.

**Comment by pwharff** [comment]
Okay, I've narrowed it down to these 2 queries: 

`RESULTS 1` query does not work:

```
    this.apollo
      .watchQuery<any>(this.queryOptions)
      .valueChanges.pipe(map (results => {
        console.log('++++++++ RESULTS 1 +++++++++', results)
      }));
```

`RESULTS 2` query works as intended:

```
    this.apollo
      .watchQuery(this.queryOptions)
      .valueChanges.subscribe((result: any) => {
        console.log('++++++++ RESULTS 2 +++++++++', result)
      });
```

The only difference between the two is one uses `pipe` and `map` while the other uses `subscribe`. Can someone explain why `pipe` and `map` example fails?

**Comment by pwharff** [comment]
Ok, I feel like an idiot! The example using a `pipe` returns an observable, which now makes sense why I wasn't getting any results yet. 

```
    this.apollo
      .watchQuery<any>(this.queryOptions)
      .valueChanges.pipe(map (results => {
        return results;
      })).subscribe(results => {
        console.log(results)
      });
```

In which case, the second example is more applicable, unless you need to pass the observable around. Hopefully this helps someone else too.

**Comment by kamilkisiela** [comment]
I think we can close it now :)
